storage/remote/read_handler.go

```diff
@@ -47,24 +47,31 @@ type readHandler struct {
 // NewReadHandler creates a http.Handler that accepts remote read requests and
 // writes them to the provided queryable.
 func NewReadHandler(logger log.Logger, r prometheus.Registerer, queryable storage.SampleAndChunkQueryable, config func() config.Config, remoteReadSampleLimit, remoteReadConcurrencyLimit, remoteReadMaxBytesInFrame int) http.Handler {
+	waitDuration := prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace:   "prometheus",
+		Subsystem:   "api",
+		Help:        "Total number of seconds spent waiting at the query gate before remote read query execution",
+		Name:        "remote_read_wait_duration_seconds_total",
+		ConstLabels: prometheus.Labels{"name": "read_handler"},
```

Comment:
do we need this label? if the metric name is unique then we should be fine

Reply:
Sounds good, removed it.